File: test/cases/record/test_record_param_value.pas
program test;
type
  person = record
    age: integer;
    score: real
  end;
var
  p: person;
  result: integer;

function getAge(p: person): integer;
begin
  getAge := p.age
end;

begin
  p.age := 25;
  p.score := 95.5;
  result := getAge(p);
  write(result)
end.

File: test/cases/record/test_record_param_value.c
#include <stdio.h>
typedef struct {
    int age;
    float score;
} person;
person p;

int result;


int getage(person p);

int getage(person p) {
    int _retval;
    _retval = p.age;

    return _retval;
}


int main(void) {
    p.age = 25;
    p.score = 95.5;
    result = getage(p);
    printf("%d", result);

    return 0;
}
